lib/node.js

```diff
@@ -973,17 +985,26 @@ declare module "fs" {
   declare function mkdtempSync(prefix: string): string;
   declare function readdir(
     path: string,
-    options: string | { encoding: string, ... },
+    options: string | { encoding?: string, withFileTypes?: false, ... },
     callback: (err: ?ErrnoError, files: Array<string>) => void
   ): void;
+  declare function readdir(
+    path: string,
+    options: { encoding?: string, withFileTypes?: true, ... },
```

Comment:
does `withFileTypes: true` need to be non-optional here? if i understand correctly, it's required to get an array of `Dirent`

Reply:
Thanks for the review @mroch! It makes sense what you are saying - I updated the PR. Could you please check again?